app-catalog/src/api/charts.tsx

```diff
@@ -1,23 +1,39 @@
 import { PAGE_OFFSET_COUNT_FOR_CHARTS } from '../components/charts/List';
 
-export function fetchChartsFromArtifact(
+export async function fetchChartsFromArtifact(
   search: string = '',
+  verified: boolean,
   category: { title: string; value: number },
   page: number,
   limit: number = PAGE_OFFSET_COUNT_FOR_CHARTS
 ) {
+  // note: we are currently defaulting to search by verified and official as default
+
   if (!category || category.value === 0) {
-    return fetch(
+    const response = await fetch(
       `https://artifacthub.io/api/v1/packages/search?kind=0&ts_query_web=${search}&sort=relevance&facets=true&limit=${limit}&offset=${
         (page - 1) * limit
-      }`
-    ).then(response => response.json());
+      }&verified_publisher=${verified}`
+    );
+
+    const total = response.headers.get('pagination-total-count');
+    const dataResponse = await response.json();
+
+    return { dataResponse, total };
+  } else {
```

Comment:
these two branches `if` `else` look almost identical, can you make it without `if`?